kahuna/public/js/search/query.js

```diff
@@ -149,7 +149,9 @@ query.controller('SearchQueryCtrl', [
     $scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
         storage.setJs("isNonFree", ctrl.filter.nonFree ? ctrl.filter.nonFree : false, true);
 
-        filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
+        if (!(filter.uploadedBy &&  filter.uploadedBy !== ctrl.user.email && filter.uploadedByMe !== true)) {
+            filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
+         }
```

Comment:
Tested it out and it works great!! Though this logic ends up being quite complex; I struggled to read follow the condition. I think it's easier to follow in the inverse condition; does this logic look equivalent to you?

```suggestion
        const myUploadsCheckbox = filter.uploadedByMe;

        // Users should be able to follow URLs with uploadedBy set to another user's name, so only
        // overwrite if:
        //   - uploadedBy is unset, or
        //   - uploadedBy is set to their email (to allow unchecking the 'My uploads' checkbox), or
        //   - 'My uploads' checkbox is checked (overwrite other user's email with theirs).
        const shouldOverwriteUploadedBy =
            !filter.uploadedBy || filter.uploadedBy === ctrl.user.email || myUploadsCheckbox;

        if (shouldOverwriteUploadedBy) {
            filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
        }
```

Reply:
@andrew-nowak Thanks for the review, yup that looks better and more readable, will test now  it and update the PR